pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -94,8 +95,13 @@ func getClusterName(ctx context.Context, data *clusterNameData, hostname string)
 					// try the next cloud provider
 					continue
 				}
+				// if the clustername is valid but contains a "_" in the middle of it, we will replace it later such that
+				// to make it valid to RFC1123.
 				if clusterName != "" {
 					log.Infof("Using cluster name %s auto discovered from the %s API", clusterName, cloudProvider)
+					if strings.HasSuffix(clusterName, "-") || strings.HasSuffix(clusterName, "_") {
+						log.Errorf("Registering an invalid clusterName as they are not allowed to end with `_` or `-`")
```

Comment:
If we trim trailing dashes now, is this error still necessary?

Reply:
No, I can remove it if we go with above. Thanks for the catch. I suggest to wait to see what the others think on above way. If we decide to go with it I will remove it. 👍